### PR TITLE
Admin action for forcing synchronization of exported API

### PR DESCRIPTION
**Pull request by jonasfj**
I hope we'll never need to manually run this, but having the button is probably a good idea.

### CODE REVIEW

**Review by isoos — APPROVED** [review]

**Review by sigurdm — COMMENTED** [review]
Is it worth creating a test of this somehow, given that we will most likely only exercise this in a state of problems...

File: app/lib/admin/actions/exported-api-sync.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    if (syncAll) {
+      await apiExporter!.synchronizeExportedApi(forceWrite: forceWrite);
+      return {
+        'packages': 'ALL',
+        'forceWrite': forceWrite,
+      };
+    } else {
+      final packages = packagesOpt.split(' ').map((p) => p.trim()).toList();
+      for (final p in packages) {
+        InvalidInputException.checkPackageName(p);
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply app/lib/admin/actions/exported-api-sync.dart:50]: Oh - sorry - you are indeed validating package names up front - nevermind

File: app/lib/admin/actions/exported-api-sync.dart [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      }
+      for (final p in packages) {
+        await apiExporter!.synchronizePackage(p, forceWrite: forceWrite);
```
Comment (sigurdm) [inline app/lib/admin/actions/exported-api-sync.dart:53]: Can we do these in parallel?

Comment (jonasfj) [reply app/lib/admin/actions/exported-api-sync.dart:53]: Let's do that in a follow up

File: app/lib/admin/actions/exported-api-sync.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/admin/actions/exported-api-sync.dart:50]: This is rejection of the request without lookups, it's cheap.

**Review by sigurdm — APPROVED** [review]

**Pull request merged**